_posts/2019-08-02-program-structure-and-composability.md

```diff
@@ -402,14 +401,17 @@ func main() {
     // before, redis.NewConn can't parse command-line.
     barRedis := redis.NewConn(cmpBar, "127.0.0.1:6379")
 
-    // If the command-line is parsed here, then how can fooRedis and barRedis
-    // have been created yet? It's only _after_ this point that `fooRedis` and
-    // `barRedis` could possibly be usable.
+    // It is only after all components have been instantiated that the
+    // command-line arguments can be parsed
     mcfg.Parse()
 }
 ```
 
-We will solve this problem in the next section.
+While this solves our argument parsing problem, fooRedis and barRedis are not
+usable yet because the actual connections have not been made. This is a classic
+chicken and the egg problem. The func `redis.NewConn` needs to make a connection
+which it cannot do until ~after~ `mcfg.Parse()` is  called. We will solve this
```

Comment:
`~` does ~strikethrough~, I think you want `_italics_` right?

Also I'd like to make the sentence a bit more wordy, to give both sides of the chicken/egg problem:

> The func `redis.NewConn` needs to make a connection which it cannot do until _after_ `mcfg.Parse` is  called, but `mcfg.Parse` cannot be called until after `redis.NewConn` has returned.